tests/Aspire.NATS.Net.Tests/AspireNatsClientExtensionsTests.cs

```diff
@@ -97,8 +113,111 @@ public void ConnectionNameWinsOverConfigSection(bool useKeyed)
             host.Services.GetRequiredKeyedService<INatsConnection>("nats") :
             host.Services.GetRequiredService<INatsConnection>();
 
-        Assert.Equal(ConnectionString, connection.Opts.Url);
+        Assert.Equal(_connectionString, connection.Opts.Url);
         // the connection string from config should not be used since it was found in ConnectionStrings
         Assert.DoesNotContain("unused", connection.Opts.Url);
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task AddNatsClient_HealthCheckShouldBeRegisteredWhenEnabled(bool useKeyed)
+    {
+        var key = DefaultConnectionName;
+        var builder = CreateBuilder(_connectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(key, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = true;
+            });
```

Comment:
```suggestion
    public async Task AddNatsClient_HealthCheckShouldBeRegisteredByDefault(bool useKeyed)
    {
        var key = DefaultConnectionName;
        var builder = CreateBuilder(_connectionString);

        if (useKeyed)
        {
            builder.AddKeyedNatsClient(key);
        }
        else
        {
            builder.AddNatsClient(DefaultConnectionName);
```

I think this would be a more appropriate test.